lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -306,6 +315,9 @@ def _update_server(self):
                     server_type=self.client.server_types.get_by_name(server_type),
                     upgrade_disk=self.module.params.get("upgrade_disk"),
                 ).wait_until_finished(timeout)
+                if state != "stopped":
+                    self.start_server()
```

Comment:
in case of `state: present` with a powered-off server it would be started here.
I would remember the server state before force-stop it and also extend the condition with it:

~~~diff
-                if state != "stopped":
+                if state != "stopped" and previous_state == Server.STATUS_RUNNING:
                     self.start_server()
~~~